src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicSafeHandle.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using Microsoft.Quic;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal abstract class MsQuicSafeHandle : SafeHandle
```

Comment:
Extracted some similar code from all our `SafeHandle` implementations. We don't actually need `SafeHandle`s atm since we're working directly with native pointers in p/invokes. But we discussed using reference counting for stream-connection lifetime relation so I'm keeping this in for now.